pkg/channel/distributed/controller/util/compare.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis/duck"
+)
+
+// CheckDeploymentChanged Modifies An Existing Deployment With New Fields (If Necessary)
+// Returns True If Any Modifications Were Made
+func CheckDeploymentChanged(logger *zap.Logger, existingDeployment, newDeployment *appsv1.Deployment) (*appsv1.Deployment, bool) {
+
```

Comment:
minor nit/suggestion - i would rename "existingDeployment" to "oldDeployment" in this context - the utility function doesn't know if it's "existing in k8s" or what (or shouldn't anyways ; )

Reply:
I agree; I replaced other "existing" with "old" in this file where appropriate